### PR TITLE
DX: add `YieldFromArrayToYieldsFixer` to `PhpCsFixer` set

### PR DESCRIPTION
**Pull request by kubawerlos**
Obviously not reviewable, to be updated after https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7114.

### CODE REVIEW

File: tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php [hunk 1]
```diff
@@ -38,98 +38,107 @@ public function testFixEchoToPrint(string $expected, ?string $input = null): voi
...
+            '<?php
                 print("test");
                 ',
-            ],
-            // `echo` can take multiple parameters (although such usage is rare) while `print` can take only one argument,
-            // @see https://php.net/manual/en/function.echo.php and @see https://php.net/manual/en/function.print.php
-            [
-                '<?php
+        ];
+        // `echo` can take multiple parameters (although such usage is rare) while `print` can take only one argument,
```
Comment (keradus) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:58]:
```suggestion
        ];

        // `echo` can take multiple parameters (although such usage is rare) while `print` can take only one argument,
```

File: tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php [hunk 2]
```diff
@@ -151,42 +160,45 @@ public function testFixPrintToEcho(string $expected, ?string $input = null): voi
...
+            '<?php
                 ($some_var) ? print "true" : print "false";
                 ',
-            ],
-            // echo has no return value while print has a return value of 1 so it can be used in expressions.
-            // https://www.w3schools.com/php/php_echo_print.asp
-            [
-                '<?php
+        ];
+        // echo has no return value while print has a return value of 1 so it can be used in expressions.
```
Comment (keradus) [inline tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:186]:
```suggestion
        ];

        // echo has no return value while print has a return value of 1 so it can be used in expressions.
```

looks like missing edge case for Fixer. maybe worth to handle it in fixer directly and regenerate this PR?

File: tests/Fixer/Operator/TernaryToNullCoalescingFixerTest.php [hunk 3]
```diff
@@ -35,119 +35,163 @@ public function testFix(string $expected, ?string $input = null): void
...
-                '<?php $x = isset($a) ? $a : null;',
-            ],
-            'Common fix case (II).' => [
-                '<?php $x = $a[0] ?? 1;',
-                '<?php $x = isset($a[0]) ? $a[0] : 1;',
-            ],
-            'Minimal number of tokens case.' => [
-                '<?php
+        yield // Do not fix cases.
+        ['<?php $x = isset($a) ? $a[1] : null;'];
```
Comment (keradus) [inline tests/Fixer/Operator/TernaryToNullCoalescingFixerTest.php:39]:
```suggestion
        // Do not fix cases:
        yield ['<?php $x = isset($a) ? $a[1] : null;'];
```

File: tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php [hunk 4]
```diff
@@ -151,42 +161,46 @@ public function testFixPrintToEcho(string $expected, ?string $input = null): voi
...
+        yield [
+            '<?php
                 echo("test");
                 ',
-            ],
-            // https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/1502#issuecomment-156436229
-            [
-                '<?php
+        ];
+        // https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/1502#issuecomment-156436229
```
[the remark this answers is not in the document]
Comment (kubawerlos) [reply tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:180]: Yes, edge case, but not for this fixer, but for the `BlankLineBeforeStatementFixer`, and it is already present: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.22.0/tests/Tokenizer/TokensTest.php#L421

I cannot comprehend why you hate single-responsibility principle so much. Why can we go with PRs that are doing some thing and leave "and this and this too" for separate PRs?

[the remark this answers is not in the document]
Comment (Wirone) [reply tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:180]: @keradus is [an uncompromising reviewer](https://twitter.com/keradus), didn't you know 😝?

[the remark this answers is not in the document]
Comment (keradus) [reply tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:180]: For my visibility, it's not that we were aware of this issue and we put it out of scope for initial PR - it rather looks for me that we simply overlooked this case initially.

I didn't see connection to other fixers so I simply _asked_ if this is the case we overlooked.

If you identify this as part of other rule responsibility - awesome. Could be we want also cover it one day, together with priority test for it.

[the remark this answers is not in the document]
Comment (kubawerlos) [reply tests/Fixer/Alias/NoMixedEchoPrintFixerTest.php:180]: Priority test is already there, we would need only the fix (feature?) for `BlankLineBeforeStatementFixer`.

**Review by keradus — APPROVED** [review]
